llvm/test/CodeGen/AIE/aie2/ra/waw_reg_renaming_loop.mir

```diff
@@ -0,0 +1,292 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 4
+# NOTE: Example file for Write After Write Register Renaming in Loop test
+#
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+
+# RUN: llc -mtriple=aie2 -verify-machineinstrs --run-pass=greedy,aie-wawreg-rewrite,virtregrewriter %s -o - | FileCheck %s
+#
+---
+name:            test_base
+alignment:       16
+legalized:       true
+tracksRegLiveness: true
+body:             |
+  ; CHECK-LABEL: name: test_base
+  ; CHECK: bb.0.entry:
+  ; CHECK-NEXT:   successors: %bb.1(0x80000000)
+  ; CHECK-NEXT:   liveins: $d0, $d2, $p0, $p1, $p2, $r0, $d1_3d
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   LoopStart $r0
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT: bb.1:
+  ; CHECK-NEXT:   successors: %bb.1(0x40000000), %bb.2(0x40000000)
+  ; CHECK-NEXT:   liveins: $d0, $d2, $p0, $p1, $p2, $d1_3d
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   renamable $p3 = COPY $p0
+  ; CHECK-NEXT:   renamable $d3 = COPY $d0
+  ; CHECK-NEXT:   $wl0, dead $p3, dead $dc3 = VLDA_2D_dmw_lda_w killed $p3, $d3
+  ; CHECK-NEXT:   renamable $p3 = COPY $p1
+  ; CHECK-NEXT:   renamable $d3_3d = COPY $d1_3d
+  ; CHECK-NEXT:   $wl2, dead $p3, dead $dc3, dead $dc7 = VLDA_3D_dmw_lda_w killed $p3, $d3_3d
+  ; CHECK-NEXT:   renamable $x1, dead renamable $l4 = VMAX_LT_D8 renamable $x0, renamable $x2, implicit $crvaddsign
+  ; CHECK-NEXT:   renamable $p3 = COPY $p2
+  ; CHECK-NEXT:   renamable $d3 = COPY $d2
+  ; CHECK-NEXT:   dead $p3, dead $dc3 = VST_2D_dmw_sts_w killed $wl1, killed $p3, $d3
+  ; CHECK-NEXT:   PseudoLoopEnd <mcsymbol .L_1120>, %bb.1
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT: bb.2:
+  ; CHECK-NEXT:   PseudoRET implicit $lr
+  bb.0.entry:
+  successors: %bb.1
+  liveins: $p0, $p1, $p2, $d0, $d1_3d, $d2
+  liveins: $r0
```

Comment:
Nit: Can you write a single `liveins` line?

Reply:
fixed